processing/src/main/java/io/druid/query/aggregation/HasDependentAggFactories.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation;
+
+import java.util.Map;
+
+public interface HasDependentAggFactories
+{
+  void setDependentAggFactories(Map<String, AggregatorFactory> aggFactoryMap);
```

Comment:
As long as a post aggregator is recognized to have the dependent aggregator factories of depedent fields in [here](https://github.com/druid-io/druid/pull/2883/files#diff-e0f622a1065b162b42516125d5438a52R65), and the post aggregator gets to have depedent aggregator factories injected, I would think that the post aggregator can extend a abstract class or implements an interface.
Do you mean introducing a abstract class and make `FinalizingFieldAccessPostAggregator` extend it? I agree with you as well. It would be better let the abstract class be in the same file as `FinalizingFieldAccessPostAggregator`